modules/lm-coursier/src/main/scala/lmcoursier/internal/ArtifactsRun.scala

```diff
@@ -38,53 +38,51 @@ object ArtifactsRun {
         else
           ""
 
-      ThreadUtil.withFixedThreadPool(params.parallel) { pool =>
-
-        coursier.Artifacts()
-          .withResolutions(params.resolutions)
-          .withArtifactTypes(Set(Type.all))
-          .withClassifiers(params.classifiers.getOrElse(Nil).toSet)
-          .withClasspathOrder(params.classpathOrder)
-          .addExtraArtifacts { l =>
-            if (params.includeSignatures)
-              l.flatMap(_._3.extra.get("sig").toSeq)
-            else
-              Nil
-          }
-          .addTransformArtifacts { artifacts =>
-            if (params.missingOk)
-              artifacts.map {
-                case (dependency, publication, artifact) =>
-                  (dependency, publication, artifact.withOptional(true))
+      coursier.Artifacts()
+        .withResolutions(params.resolutions)
+        .withArtifactTypes(Set(Type.all))
+        .withClassifiers(params.classifiers.getOrElse(Nil).toSet)
+        .withClasspathOrder(params.classpathOrder)
+        .addExtraArtifacts { l =>
+          if (params.includeSignatures)
+            l.flatMap(_._3.extra.get("sig").toSeq)
+          else
+            Nil
+        }
+        .addTransformArtifacts { artifacts =>
+          if (params.missingOk)
+            artifacts.map {
+              case (dependency, publication, artifact) =>
+                (dependency, publication, artifact.withOptional(true))
+            }
+          else
+            artifacts
+        }
+        .withCache(
+          params
+            .cache
+            .withPool(params.cache.pool)
```

Comment:
No need to call that, the `pool` field of `params.cache` is already `params.cache.pool`.